zasm/include/zasm/program/program.hpp

```diff
@@ -263,6 +263,20 @@ namespace zasm
         /// <returns>Node for which the label is bound to or null if the label is invalid or not bound</returns>
         Node* getNodeForLabel(const Label& label);
 
+        /// <summary>
+        /// Returns the current assigned label name, returns nullptr if the label is invalid or has no name.
+        /// </summary>
+        /// <param name="label">Label</param>
+        /// <returns>Label name or nullptr</returns>
+        const char* getLabelName(const Label& label) const noexcept;
+
+        /// <summary>
+        /// Assigns a name to a label.
+        /// </summary>
+        /// <param name="label">Label</param>
+        /// <param name="name">The new name, pasing nullptr will clear the name</param>
```

Comment:
Should probably clarify that the string is non-owning and will be copied.

Reply:
Done, I will probably have to go over some other code that receives `const char*` and clarify that and maybe re-work the use of strings.